### PR TITLE
chore: add virtualization utilities for ExploreTree

### PR DESCRIPTION
**Pull request by almeidabbm**
### Description:
Implements a virtualized tree flattening algorithm for the Explorer component to improve performance with large datasets. The implementation includes:

- A new `flattenTreeForVirtualization` function that converts the hierarchical tree structure into a flat list of items for virtual rendering
- Support for different item types (table headers, section headers, tree nodes, missing fields, empty states)
- Proper handling of expanded/collapsed states for tables and groups
- Search functionality that automatically expands relevant sections
- Comprehensive test coverage for all edge cases

This approach optimizes rendering by only creating DOM elements for visible items, significantly improving performance when exploring tables with many dimensions and metrics.

### CODE REVIEW

**Comment by almeidabbm** [comment]
* **#17653** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17653?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17651** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17651?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17646** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17646?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17620** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17620?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17619** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17619?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17618** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17618?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17617** <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17617?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/lightdash/lightdash/17617?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Virtualization/flattenTree.ts [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+        itemsMap: sectionInfo.itemsMap,
+        missingCustomMetrics:
+            sectionInfo.type === TreeSection.CustomMetrics &&
+            sectionInfo.missingItems
+                ? (sectionInfo.missingItems as any[])
+                : undefined,
+        missingCustomDimensions:
+            sectionInfo.type === TreeSection.CustomDimensions &&
+            sectionInfo.missingItems
+                ? (sectionInfo.missingItems as any[])
```
Comment (pimmey) [inline packages/frontend/src/components/Explorer/ExploreTree/TableTree/Virtualization/flattenTree.ts]: Those 2x`any[]` could be `AdditionalMetric[]` and `CustomDimension[]`.

**Review by pimmey — APPROVED** [review]
Sweet utils and tests! Just 1 lil TS thing.

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Virtualization/flattenTree.ts (hunk 1 above)
Comment (almeidabbm) [reply packages/frontend/src/components/Explorer/ExploreTree/TableTree/Virtualization/flattenTree.ts]: Was removing them as we speak eheh, should be good now 🙇
